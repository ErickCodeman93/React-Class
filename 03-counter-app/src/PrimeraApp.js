// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';


//Funcional Component

const PrimeraApp = ( { saludo2 = 'Hola Mundo', saludo3 , subtitulo } ) => {
	// return ( 
	// 	<div>
	// 		<h1>Hola</h1>
	// 		<p>Mi primera aplicación</p>
	// 	</div>
	// 	);
	// return ( <Fragment> 
	// 			<h1>Hola</h1>
	// 	 		<p>Mi primera aplicación</p>
	// 		</Fragment> );
	const saludo = 'Erick';

	// const info = {
	// 	'edad' : 28,
	// 	'state' : 'Edomex'
	// }

	// console.log( saludo2 );

	return ( 
		<>
			<h1>Hola { saludo }</h1>
			<br/>
			<h2>{ saludo2 }</h2>
			{/* <p>Mi primera aplicación</p> */}
			<p>{ subtitulo }</p>
			{/* <pre> { JSON.stringify( info, null, 3 ) } </pre> */}
		</>
	);
} //end component

PrimeraApp.propTypes = {
	saludo2 : PropTypes.string.isRequired
}

//Este sirve para que el desarrollador desde la consola verifique con que propiedades cuenta el
// componente
PrimeraApp.defaultProps = {
	subtitulo: 'Soy un subtitulo',
	saludo3: ''
}

export default PrimeraApp;