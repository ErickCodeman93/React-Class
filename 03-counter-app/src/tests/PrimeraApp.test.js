import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';
// import { render } from '@testing-library/react'

describe('Pruebas en <PrimeraApp />', () => {

	// test('Debe mostrar el tÃ­tulo "Hola, Soy Goku"', () => {
		
	// 	const saludo = 'Hola, Soy Goku';
	// 	const { getByText } = render( <PrimeraApp saludo2={ saludo }/> );

	// 	expect( getByText( saludo ) ).toBeInTheDocument();
	// 	// wrapper.getByText();
	// });
	
	test('debe mostrar <primerApp /> correctamente ', () => {

		const saludo = 'Hola, Soy Goku';
		const wrapper = shallow( <PrimeraApp saludo2={ saludo }/> );
		expect( wrapper ).toMatchSnapshot();
	});

	test('debe mostrar un subtitulo con props', () => {
		
		const saludo = 'Hola, Soy Goku';
		const subtitulo = 'Hola soy un subtitulo';

		const wrapper = shallow( 
			<PrimeraApp 
				saludo2={ saludo }
				subtitulo={ subtitulo }
			/> 
		);

		const textoParrafo = wrapper.find( 'p' ).text();
		// console.log( textoParrafo );

		expect( textoParrafo ).toBe( saludo );
	});
	
	
})
