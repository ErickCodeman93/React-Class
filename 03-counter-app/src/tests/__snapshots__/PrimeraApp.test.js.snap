// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <primerApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola 
    Erick
  </h1>
  <br />
  <h2>
    Hola, Soy Goku
  </h2>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
